### PR TITLE
Add NodeInterface pages

### PR DESCRIPTION
**Pull request by TucksonDev**
This PR improves the information available about the NodeInterface. It creates a concept page and a reference page, autogenerating the table that contains the methods available in the NodeInterface. It also changes all previous links going to the short section in the L2 gas page to the new concept or reference page.

[Concept page](https://nitro-docs-git-create-nodeinterface-pages-offchain-labs.vercel.app/for-devs/concepts/nodeinterface)
[Reference page](https://nitro-docs-git-create-nodeinterface-pages-offchain-labs.vercel.app/for-devs/dev-tools-and-resources/nodeinterface)

### CODE REVIEW

File: arbitrum-docs/arbos/gas.mdx [hunk 1]
```diff
@@ -27,21 +27,6 @@ When a transaction schedules another, the subsequent transaction's execution [wi
 
 Because a call to [`redeem`](/for-devs/dev-tools-and-resources/precompiles.mdx#arbretryabletx) donates all of the call's gas, doing multiple requires limiting the amount of gas provided to each subcall. Otherwise the first will take all of the gas and force the second to necessarily fail irrespective of the estimation's gas limit.
 
-Gas estimation for Retryable submissions is possible via [`NodeInterface.sol`][node_interface_link] and similarly requires the auto-redeem attempt succeed.
+Gas estimation for Retryable submissions is possible via the [NodeInterface](/for-devs/dev-tools-and-resources/nodeinterface.mdx) and similarly requires the auto-redeem attempt succeed.
```
Comment (mahsamoosavi) [inline arbitrum-docs/arbos/gas.mdx:30]:
```suggestion
Gas estimation for Retryable submissions is possible via the [NodeInterface](/for-devs/dev-tools-and-resources/nodeinterface.mdx) and similarly requires the auto-redeem attempt to succeed.
```

File: arbitrum-docs/arbos/l1-pricing.mdx [hunk 2]
```diff
@@ -51,6 +51,6 @@ A second term is added to the L1 Gas Basefee, based on the derivative of the sur
 
 ## Getting L1 Fee Info
```
Comment (mahsamoosavi) [inline arbitrum-docs/arbos/l1-pricing.mdx:52]:
```suggestion
## Getting L1 fee info
```

File: arbitrum-docs/for-devs/concepts/nodeinterface.mdx [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+description: A high level description of what the NodeInterface is and how it works
+reader-audience: developers who want to build on Ethereum/Arbitrum
+content-type: concept
+---
+
+import PublicPreviewBannerPartial from '../../partials/_public-preview-banner-partial.md';
+
+<PublicPreviewBannerPartial />
+
+The Arbitrum Nitro software includes a special `NodeInterface` contract available at address `0xc8` that is only accessible via RPCs (it's not actually deployed on-chain, and thus can't be called by smart contracts). The way it works is that the node uses geth's [`InterceptRPCMessage`](https://github.com/OffchainLabs/go-ethereum/blob/@goEthereumCommit@/internal/ethapi/api.go#L1034) hook to detect messages sent to the address `0xc8`, and swaps out the message it's handling before deriving a transaction from it.
```
Comment (mahsamoosavi) [inline arbitrum-docs/for-devs/concepts/nodeinterface.mdx:13]:
```suggestion
The Arbitrum Nitro software includes a special `NodeInterface` contract available at address `0xc8` that is only accessible via RPCs (it's not actually deployed on-chain, and thus can't be called by smart contracts). The way it works is that the node uses Geth's [`InterceptRPCMessage`](https://github.com/OffchainLabs/go-ethereum/blob/@goEthereumCommit@/internal/ethapi/api.go#L1034) hook to detect messages sent to the address `0xc8`, and swaps out the message it's handling before deriving a transaction from it.
```

**Review by mahsamoosavi — CHANGES_REQUESTED** [review]
a few suggestions, otherwise looks 🔥

**Comment by TucksonDev** [comment]
Thanks @mahsamoosavi ! 🙏 
Btw, those pages under the `arbos` folder still need a deep rework (my intention was just to change the links pointing to the new NodeInterface page).

**Review by symbolpunk — APPROVED** [review]
Very cool 😎

**Review by mahsamoosavi — APPROVED** [review]

**Pull request merged**